src/components/shared/table/TractionTable.vue

```diff
@@ -0,0 +1,372 @@
+                  <traction-table-column :classes="`border-0`">
+                    <slot :name="`row-details`" v-bind="row" />
+                  </traction-table-column>
+                </traction-table-row>
+              </template>
+            </tbody>
+          </table>
+        </div>
+      </div>
+    </div>
+  </flagged-feature>
+</template>
+<script>
+import TractionSortIcon from '@/components/shared/icons/TractionSortIcon'
+import BTableWrapper from '@/components/shared/table/BTableWrapper'
+import { alphaNumericSortDefault, flattenObject } from '@/lib/DataHelpers'
+import { within } from '@/lib/propValidations'
+
+export default {
+  name: 'TractionTable',
+  components: { TractionSortIcon, BTableWrapper },
+  inheritAttrs: false,
+  props: {
+    //attribute name to represent this component for testing, if given
+    dataAttribute: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Header fields
+     * key - Key field of header column
+     * label - Text to display on header
+     * formatter - Formatter function if any customization required to display the header label
+     */
+    fields: {
+      type: Array,
+      default: () => [{ key: '', label: '', formatter: () => {} }],
+    },
+    /**
+     * Data to display in table
+     * This should be either an object array having field names same as header field keys or a string array
+     */
+    items: {
+      type: Array,
+      required: false,
+      default: () => [],
+    },
+    /**
+     * current sort-by column field
+     */
+    sortBy: {
+      type: String,
+      required: false,
+      default: '',
+    },
+    /**
+     * Primary key field to uniquely identify a row.
+     * This can be particularly useful in cases where table is used with 'showDetails' functionality
+     * (which allows the user to display extra information),
+     * This key make sure that open/close row status will not be lost while sorting data
+     *
+     */
+    primaryKey: {
+      type: String,
+      required: false,
+      default: 'id',
+    },
+    /**
+     * Specifies whether the table rows are selectable
+     */
+    selectable: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    /**
+     * Single/multiple row selection?
+     */
+    selectMode: {
+      type: String,
+      required: false,
+      default: 'single',
+      validator: () => within('single', 'multiple'),
+    },
+    /**
+     * This allows to use table header with any slot component
+     * 'fields' is the only mandatory prop and 'items' doesn't have any effect in this case
+     */
+    simple: {
+      type: Boolean,
+      required: false,
+    },
+  },
+  data() {
+    //Create 'row' data based on initial data passed in through 'items' prop
+    const rows = this.generateRowData()
+    return {
+      rows: rows,
+      //TODO: Ascending property to be passed as a prop
```

Comment:
Yes, I think so. Now, by default the sort type is ascending, but in case we need to display a table column (sortBy) with descending by default, such a prop might be needed. But, haven't seen any use case yet.